### ISSUE TITLE
I don't see any working function on windows canary

### ISSUE THREAD

**Issue by SWATOPLUS** [oven-sh/bun#7197]
### What version of Bun is running?

1.0.12-canary.13+3ca2d8ae5

### What platform is your computer?

Microsoft Windows NT 10.0.22621.0 x64

### What steps can reproduce the bug?

I am trying to do something with bun on windows but it doesn't work.

I have downloaded bun windows canary exe-file and put it to `d:/bun-canary` folder. I also created `index.mts` file at this folder with single line `console.log('Hello world');`

So, I run `bun init`

![image](https://github.com/oven-sh/bun/assets/13176763/1ec21d9b-a5f7-41f4-bc44-3a19603f3552)

I am not able to just press enter two times to apply defaults, because the program stucks. If I try to provide name more then 3 symbols the program stucks. So I typed `abc` as module name and `index.mts` as entry point.

`package.json` file has been generated, but it has incorrect structure.

![image](https://github.com/oven-sh/bun/assets/13176763/1642f70a-641a-4b5f-ba3f-5207b8bb1333)

Ok, then I try to run `main.mts`

```
d:\bun-canary>bun run index.mts
error: missing script "index.mts"

d:\bun-canary>bun run ./index.mts
error: missing script "./index.mts"

d:\bun-canary>bun run .\index.mts
error: missing script ".\index.mts"

d:\bun-canary>bun run d:/bun-canary/index.mts
d: [1/1]
error: FailedToOpenSocket downloading package manifest d:
error: Cannot find module "d:/bun-canary/index.mts" from "bun:main"

d:\bun-canary>bun run d:\bun-canary\index.mts
d: [1/1]
error: FailedToOpenSocket downloading package manifest d:
error: Cannot find module "d:/bun-canary/index.mts" from "bun:main"
```

Ok, let's try something else
```
d:\bun-canary>bun install
install is not supported on Windows yet, sorry!!
```
It's ok, because I see honest message, that it is not supported.

```
d:\bun-canary>bun x cowsay hello
# many errors here
```
I have reported about this two weeks ago (https://github.com/oven-sh/bun/issues/6910), but issue has been closed because it isn't suppotred like `bun install`

So, let's just add honest message about this like in `bun install`

```
d:\bun-canary>bun build .\index.mts
error: Cannot read file "d:/bun-canary\": Unexpected
```
Still something going wrong.

```
d:\bun-canary>bun test
bun test v1.0.12-canary.13 (3ca2d8ae)
No tests found!
Tests need ".test", "_test_", ".spec" or "_spec_" in the filename (ex: "MyApp.test.ts")
[7.00ms]
```
It always says no tests found, even if I put some test files. `demo.spec.ts`:
![image](https://github.com/oven-sh/bun/assets/13176763/1836551c-8736-467d-8c4e-9194174f3bdb)

```
d:\bun-canary>bun -e "console.log('Hello, JavaScript!')"
Hello, JavaScript!

error: Cannot read file "d:/bun-canary\": Unexpected
```

Oh, I have found something that works, but still some error happens.

```
d:\bun-canary>bun repl
bun-repl [4/5] error: InvalidArgument extracting tarball for bun-repl@swc/core [8/20] error: InvalidArgument extracting tarball for @swc/core@swc/types [34/36] error: InvalidArgument extracting tarball for @swc/types@swc/core-win32-x64-msvc [36/37] error: InvalidArgument extracting tarball for @swc/core-win32-x64-msvc@swc/counter [38/38] error: InvalidArgument extracting tarball for @swc/counterSaving lockfile... error: failed to write lockfile
EUNKNOWN: An unknown error occurred (write())
```

Oh, okay last try, I have patched package,json to be correct
![image](https://github.com/oven-sh/bun/assets/13176763/fcfea09a-7ed4-4987-9417-4ee349e402d8)

```
d:\bun-canary>bun run demo
error: missing script "demo"

d:\bun-canary>pnpm demo

> abc@ demo D:\bun-canary
> echo 42

42

d:\bun-canary>
```

### What is the expected behavior?

I want to see any roadmap of windows support. What features is expected to be implemented at first windows release? (That should be happed in the past (2023.11.14 in a same day with C# 12 and .NET 8)).

I also want to see honest messages that some feature is not expected to work, and I should not submit a bug report.

Just provide this information in `bun --help` / `bun`

![image](https://github.com/oven-sh/bun/assets/13176763/c3a57a8f-83dc-4377-b4e8-714b53b91fe3)

Please do this:
1. Provide a list of features that are expecting in first windows release.
2. Provide a list of feature statuses: 
* what is expected to work (with list of known issues)
* what is not implemeted yet (bugs for this items should not be reported).
4. Write a honest message that feature in not supported in first windows release or not ready yet. You can also create tracking issues in github for each feature and print number of this issue in message.
5. Hire a PM. )) I can provide some candidates that will work for free, just give them direct contacts of developers and provide your vision of roadmap.
6. Feel free to create `up for grubs` tasks that could be implemented without deep diving in bun arcitecture, like some isolated parts with OS iteraction (e.g. you have some class that uses temp folder in linux, so we need windows equivalent)

### What do you see instead?

Looks like bun team "забила хуй" on windows implementation.

**Comment by Electroid** [comment]
There is a reason we have not published a "versioned" (e.g. 1.1) release of Bun on Windows, and that's because we're still working on it. Thanks for digging in and writing up these issues. @paperdave is working on it.

**Comment by paperclover** [comment]
you're using an out of date build (confusing because you did pick the latest build off main, sorry for this). almost every issue you have pointed out is fixed on #6938, minus bun install/bunx related stuff. duplicate of #7155. see also: #43

> Write a honest message that feature in not supported in first windows release or not ready yet.

in my pr the documentation will be updated to note this build is unstable. the current messaging in the docs is bad
